File: sqlite-android-compiler/src/main/java/com/alecstrong/sqlite/android/model/Table.java
package com.alecstrong.sqlite.android.model;

import com.alecstrong.sqlite.android.SqliteCompiler;
import com.squareup.javapoet.ClassName;
import com.squareup.javapoet.TypeName;
import java.util.ArrayList;
import java.util.List;

public final class Table<T> extends SqlElement<T> {
  private final String packageName;
  private final String name;
  private final String interfaceName;
  private final List<Column<T>> columns = new ArrayList<Column<T>>();
  private final boolean isKeyValue;

  public Table(String packageName, String interfaceName, String name, T originatingElement,
      boolean isKeyValue) {
    super(originatingElement);
    this.packageName = packageName;
    this.interfaceName = interfaceName;
    this.name = name;
    this.isKeyValue = isKeyValue;
  }

  public void addColumn(Column<T> column) {
    columns.add(column);
  }

  /*
   * Compiler methods
   */

  public String getPackageName() {
    return packageName;
  }

  public List<Column<T>> getColumns() {
    return columns;
  }

  private String modelName() {
    return interfaceName;
  }

  public String sqlTableName() {
    return name;
  }

  public String interfaceName() {
    return SqliteCompiler.interfaceName(modelName());
  }

  public String mapperName() {
    return modelName() + "Mapper";
  }

  public String marshalName() {
    return modelName() + "Marshal";
  }

  public TypeName interfaceType() {
    return ClassName.get(packageName, interfaceName());
  }

  public boolean isKeyValue() {
    return isKeyValue;
  }
}

File: sqlite-android-compiler/src/main/kotlin/com/squareup/sqlite/android/model/Table.kt
/*
 * Copyright (C) 2016 Square, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.squareup.sqlite.android.model

import com.squareup.javapoet.ClassName
import com.squareup.javapoet.TypeName
import com.squareup.sqlite.android.SqliteCompiler
import java.util.ArrayList

class Table<T>(val packageName: String, internal val name: String, val sqlTableName: String,
    originatingElement: T, val isKeyValue: Boolean) : SqlElement<T>(originatingElement) {
  val columns = ArrayList<Column<T>>()
  val interfaceName: String
    get() = SqliteCompiler.interfaceName(name)
  val interfaceType: TypeName
    get() = ClassName.get(packageName, interfaceName)
}
